storagemarket/deal_execution.go

```diff
@@ -601,6 +602,15 @@ func openReader(filePath string, pieceSize abi.UnpaddedPieceSize) (io.ReadCloser
 }
 
 func (p *Provider) indexAndAnnounce(ctx context.Context, pub event.Emitter, deal *types.ProviderDealState) *dealMakingError {
+	// If this is Curio sealer then we should wait till sector finishes sealing
+	if p.config.Curio {
+		// Wait for sector to finish sealing
+		err := p.trackCurioSealing(deal.SectorID)
```

Comment:
I guess now that we have a curio-specific path here we could start thinking about e.g. making the indexing step a curio task instead of running that here.

Reply:
Yes. That would be the next step. I will need to make some changes in PieceDirectory. We will need to add an extra API in curio-market from where this call is done. I want it to be separate from AP() call.